### PR TITLE
feat: add initial paging support to REST clients

### PR DESCRIPTION
**Pull request by software-dov**
This is a first step to bring REST paging into parity with gRPC
paging.
Subsequent PRs will modify this surface to add support for map paging.

Adds showcase Echo REST integration tests

### CODE REVIEW

File: internal/gengapic/genrest.go [hunk 1]
```diff
@@ -529,24 +529,85 @@ func (g *generator) pagingRESTCall(servName string, m *descriptor.MethodDescript
...
+	if info == nil {
+		return errors.E(nil, "method has no http info: %s", m.GetName())
+	}
+	max := "math.MaxInt32"
+	ps := "int32(pageSize)"
+
+	verb := strings.ToUpper(info.verb)
+
 	pageSizeFieldName := snakeToCamel(pageSize.GetName())
+	g.imports[pbinfo.ImportSpec{Path: "google.golang.org/protobuf/encoding/protojson"}] = true
```
Comment (noahdietz) [inline internal/gengapic/genrest.go:556]: Nit: move this import to the bottom of this function with the other utility imports

File: internal/gengapic/genrest.go [hunk 2]
```diff
@@ -529,24 +529,85 @@ func (g *generator) pagingRESTCall(servName string, m *descriptor.MethodDescript
...
+	}
+	info, err := getHTTPInfo(m)
+	if err != nil {
+		return err
+	}
+	if info == nil {
+		return errors.E(nil, "method has no http info: %s", m.GetName())
+	}
+	max := "math.MaxInt32"
+	ps := "int32(pageSize)"
```
Comment (noahdietz) [inline internal/gengapic/genrest.go:551]: Why add these to variables? It doesn't save much, using literal values in the print statements is fine?

File: internal/gengapic/genrest.go [hunk 3]
```diff
@@ -529,24 +529,85 @@ func (g *generator) pagingRESTCall(servName string, m *descriptor.MethodDescript
...
 	inSpec, err := g.descInfo.ImportSpec(inType)
 	if err != nil {
 		return err
 	}
 
-	// outSpec, err := g.descInfo.ImportSpec(outType)
-	// if err != nil {
-	// 	return err
-	// }
+	outSpec, err := g.descInfo.ImportSpec(outType)
```
[the remark this answers is not in the document]
Comment (software-dov) [reply internal/gengapic/genrest.go:539]: Of course

File: internal/gengapic/genrest.go (hunk 2 above)
Comment (software-dov) [reply internal/gengapic/genrest.go:551]: Copy/paste mode :P

**Review by noahdietz — APPROVED** [review]

File: showcase/echo_test.go [hunk 4]
```diff
@@ -227,24 +244,26 @@ func TestPaginationWithToken(t *testing.T) {
 	str := "ab cd ef gh ij kl"
 	expected := strings.Split(str, " ")[1:]
 	req := &showcasepb.PagedExpandRequest{Content: str, PageSize: 2, PageToken: "1"}
```
Comment (vchudnov-g) [inline showcase/echo_test.go:246]: In https://github.com/googleapis/gapic-showcase/pull/806 I also added a `PagedExapndLegacy()` RPC that uses amx_results, so ideally we'd test that as well. (But that version of Showcase is not released yet, though we can do that).  I would add a TODO for now at least, and when we release Showcase (which may be after your mapped response PR and subsequent generator release!) we can come back and add the tests for this and for the mapped response, and re-release the generator. Chicken-and-egg problem!

**Review by vchudnov-g — APPROVED** [review]

File: showcase/echo_test.go [hunk 5]
```diff
@@ -267,6 +267,9 @@ func TestPaginationWithToken(t *testing.T) {
 	}
 }
 
+// TODO(dovs): the server side is not finished for PagedExpandLegacy.
+// Add tests for that when it's been merged.
```
Comment (vchudnov-g) [inline showcase/echo_test.go:271]: Well, it's been merged, just not released.

**Pull request merged**